src/ProxyKit/ForwardContext.cs

```diff
@@ -27,6 +27,8 @@ public class ForwardContext
         /// </summary>
         public HttpContext HttpContext { get; }
 
+        public TimeSpan Timeout { get => _httpClient.Timeout; set => _httpClient.Timeout = value; }
```

Comment:
@damianh , call `IHttpClientFactory.CreateClient(string name)` always return a new `HttpClient` , but `HttpMessageHandler` is reused.

See https://github.com/aspnet/Extensions/blob/8792791e713153349025f727ba54983e5aa22617/src/HttpClientFactory/Http/src/DefaultHttpClientFactory.cs#L117-L133

Reply:
Hmm, allocating a new HttpClient for each request... I'm going to have to think about that.

Reply:
Might be related to https://github.com/damianh/ProxyKit/issues/155

Reply:
Reuse HttpClient isn't a good idea, it confilits with `HttpMessageHandler` managment.

Reply:
My understanding of HttpClient via HttpClientFactory is that the lifcycle of the httpclient is independent of the underlying HttpMessageHandlers which are rebuilt.

> Manage the lifetime of HttpClientMessageHandlers to avoid the mentioned problems/issues that can occur when managing HttpClient lifetimes yourself.

https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

> If you are on .NET Core - you should use a single HttpClient directly and set SocketsHttpHandler.PooledConnectionTimeout here to an appropriate value.

https://github.com/aspnet/Extensions/issues/1345#issuecomment-480548175

It looks like a single HttpClient instance is fine, especially with new SocketsHandler sorting the network/dns behaviour issues.

@davidfowl , would appreciate your input on this :)